src/ProgressOnderwijsUtils/SingleSignOn/SsoProcessor.cs

```diff
@@ -195,12 +138,19 @@ static string[] GetAttributes(XElement assertion, string key)
                 select attribute.Value).ToArray();
         }
 
-        public static Saml20MetaData GetMetaData(IdentityProviderConfig idp, ServiceProviderConfig sp)
-            => ValidatedSaml20MetaData(idp, ConextMetaDataXml(idp, sp));
+        static readonly MemoryCache memoryCache = new MemoryCache(new MemoryCacheOptions());
 
-        static XmlDocument ConextMetaDataXml(IdentityProviderConfig idp, ServiceProviderConfig sp)
+        public static Saml20MetaData GetMetaData(IdentityProviderConfig idp, ServiceProviderConfig sp, TimeSpan expiration)

```

Comment:
I'm not sure a client can call this api correctly; there's no way in advance to know what an appropriate expiration is (other than 0). Either we should use the http mechanisms (but that's useless for surff), or isn't there some validity specified in the downloaded metadata we can use?